### ISSUE TITLE
DeploymentConfig ImageChange trigger seems to be wrong for custom images

### ISSUE THREAD

**Issue by manusa** [eclipse/jkube#904]
### Description

The following statements sets the **namespace** of the change trigger to the image user. This seems to be an issue for custom images which include the repository/user/group information.

https://github.com/eclipse/jkube/blob/682cc6f25ccf3e95e20832d1396fa3890f3a4e26/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/ImageChangeTriggerEnricher.java#L98

You can easily reproduce this with [quickstarts/maven/quarkus-customized-image](https://github.com/eclipse/jkube/tree/5e8b297a379cbe13745574a34440e3c0bcfa291c/quickstarts/maven/quarkus-customized-image).

- [x] Check if issue is reproducible in Fabric8 Maven Plugin 
- [ ] [...]

**Comment by rohanKanojia** [comment]
When I was trying to reproduce this issue, Due to image name being set to `${project.groupId}/${project.artifactId}` I bumped into https://github.com/eclipse/jkube/issues/908. I modified image name to `%g/${project.artifactId}` But noticed a difference in behavior between Fabric8 Maven Plugin and Eclipse JKube.

In Fabric8 Maven Plugin, `%g` was getting replaced by namespace which was currently configured or picked from kube config. But in case of Eclipse JKube, it was always set to `${project.groupId}`.

In ResourceMojo, there is a lateInit method which seems to be setting `docker.image.user` property when OpenShift platform is detected:

https://github.com/eclipse/jkube/blob/e2aa310a834322d5eca84f802e5830d706c47d21/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java#L81


https://github.com/eclipse/jkube/blob/e2aa310a834322d5eca84f802e5830d706c47d21/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java#L299

This property seems to be read from ImageNameFormatter, **note that what's being read is `jkube.image.user`**:

https://github.com/eclipse/jkube/blob/e2aa310a834322d5eca84f802e5830d706c47d21/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/helper/ImageNameFormatter.java#L93-L100

I think we have a typo in ResourceMojo, we should update this property to match the one being read in ImageNameFormatter. I'll create a PR to fix this.

**Comment by rohanKanojia** [comment]
According to OpenShift documentation, `from.namespace` field is optional. If we don't set this namespace is assumed to be namespace where DeploymentConfig gets applied.

![Screenshot from 2021-09-14 18-27-31](https://user-images.githubusercontent.com/13834498/133261837-e8cbdc59-3ad5-47f8-b229-72a1cdc9beff.png)

If I remove this namespace set line from ImageChangeTriggerEnricher, I can see that in `target/classes/META-INF/jkube/openshift.yml` ImageChangeTrigger's `from.namespace` is no longer set:
```yaml
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - orgeclipsejkubequickstartsmaven-quarkus-customized-image
        from:
          kind: ImageStreamTag
          name: quarkus-customized-image:latest
      type: ImageChange
```
But after issuing `mvn oc:apply`, I can see it in applied DeploymentConfig.
```yaml
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - orgeclipsejkubequickstartsmaven-quarkus-customized-image
        from:
          kind: ImageStreamTag
          name: quarkus-customized-image:latest
          namespace: rokumar-dev
        lastTriggeredImage: image-registry.openshift-image-registry.svc:5000/rokumar-dev/quarkus-customized-image@sha256:a080942a695f6c29ebe4d2b98d294bdd41124000d97ea7066ff3078c4fe07645
      type: ImageChange
```


I have tested with both `%g` as image user and also `${project.groupId}`(with a fix to #908) and it seems to be working well.

**Comment by manusa** [comment]
OK, let's remove it then.

If there's a regression we can always set it back since the correct behavior is fixed in #910

Updated documentation for 4.8 https://docs.openshift.com/container-platform/4.8/rest_api/workloads_apis/deploymentconfig-apps-openshift-io-v1.html
